File: beginner/pagina12/3162/3162.js
const fs = require("fs")
const input = fs.readFileSync("/dev/stdin", "utf8").split("\n")

function calcDistancia(x1, y1, z1, x2, y2, z2){
    return Math.sqrt(
        Math.pow(x2 - x1, 2) +
        Math.pow(y2 - y1, 2) +
        Math.pow(z2 - z1, 2)
    )
}

const n = Number(input.shift())
const naves = input.slice(0, n).map(line => line.split(' ').map(Number))

for(let i = 0; i < n; i++){
    let [nax, nay, naz] = naves[i]
    let menorDistancia = Infinity

    for(let j = 0; j < n; j++){
        if(i !== j){
            let [npx, npy, npz] = naves[j]
            let distancia = calcDistancia(nax, nay, naz, npx, npy, npz)

            if(distancia < menorDistancia){
                menorDistancia = distancia
            }
        }
    }

    if(menorDistancia > 50){ console.log("B") }
    else if (menorDistancia > 20 && menorDistancia <= 50){ console.log("M")}
    else {console.log("A")}
}

File: beginner/__3000__/3162/3162.py
import math

def calcDistancia(x1, y1, z1, x2, y2, z2):
    return math.sqrt(
        ((x2 - x1) ** 2) +
        ((y2 - y1) ** 2) +
        ((z2 - z1) ** 2)
    )


n = int(input())
naves = []

for i in range(n):
    x, y, z = list(map(int, input().split()))
    naves.append([x, y, z])

for i in range(n):
    menorDistancia = float("inf")
    nax, nay, naz = naves[i]

    for j in range(n):
        if i != j:
            npx, npy, npz = naves[j]
            distancia = calcDistancia(nax, nay, naz, npx, npy, npz)

            if distancia < menorDistancia:
                menorDistancia = distancia

    if menorDistancia > 50: 
        print("B") 
    elif 20 < menorDistancia <= 50: 
        print("M")
    else:
        print("A")